src/axom/quest/MarchingCubesAlgo.cpp

```diff
@@ -0,0 +1,614 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/quest/MarchingCubesAlgo.hpp"
+#include "axom/quest/detail/marching_cubes_lookup.hpp"
+#include "axom/mint/mesh/UnstructuredMesh.hpp"
+#include "conduit_blueprint.hpp"
+#include "axom/fmt.hpp"
+
+#ifndef __WHERE
+  #define __STRINGIZE(x) __STRINGIZE2(x)
+  #define __STRINGIZE2(x) #x
+  //!@brief String literal for code location
+  #define __WHERE \
+    __FILE__ ":" __STRINGIZE(__LINE__) "(" + std::string(__func__) + ") "
+#endif
+
+namespace axom
+{
+namespace quest
+{
+MarchingCubesAlgo::MarchingCubesAlgo(const conduit::Node& bpMesh,
+                                     const std::string& coordsetName,
+                                     const std::string& maskField)
+  : _sd()
+  , _ndim(0)
+  , _coordsetPath("coordsets/" + coordsetName)
+  , _fcnPath()
+  , _maskPath(maskField.empty() ? std::string() : "fields/" + maskField)
+  , _surfaceMesh(nullptr)
+  , _cellIdField()
+  , _domainIdField()
+{
+  _sd.reserve(conduit::blueprint::mesh::number_of_domains(bpMesh));
+  for(auto& dom : bpMesh.children())
+  {
+    _sd.emplace_back(new MarchingCubesAlgo1(dom, coordsetName, maskField));
+    if(_ndim == 0)
+    {
+      _ndim = _sd.back()->dimension();
+    }
+    else
+    {
+      SLIC_ASSERT(_ndim == _sd.back()->dimension());
+    }
+  }
+}
+
+void MarchingCubesAlgo::set_function_field(const std::string& fcnField)
+{
+  _fcnPath = "fields/" + fcnField;
```

Comment:
Should we validate `fcnField` to ensure it has the right form? Presumably alphanumeric and underscores. Not sure if it can include slashes.

Reply:
I thought our convention for naming class member variables used the "m_" prefix; e.g., "m_foo". Did we change that?

Reply:
@kennyweiss On one hand, the check on valid key names would be helpful.  On the other, I don't think it's hard to track it down from the diagnostic output when there's a bad key name.  We access Conduit and Sidre keys many places.  Maybe in `core/utilities`?

@rhornung67 I'll add the `m` before the `_`, since that probably is the Axom convention.  However, my 2 cents is that it's harder to type and doesn't add any more clarity.

Reply:
Added https://github.com/LLNL/axom/issues/1063 to discuss validating key names.